Assets/PatchKit Patcher/Scripts/Debug/DebugMenu.cs

```diff
@@ -0,0 +1,242 @@
+            int y = (Screen.height - windowHeight) / 2;
+            int popupRectY = (Screen.height - 120) / 2;
+            _rect = new Rect(x, y, windowWidth, windowHeight);
+            _popupRect = new Rect(x, popupRectY, windowWidth, 120);
+            _graphicRaycaster = FindObjectOfType<GraphicRaycaster>();
+        }
+
+        void OnGUI()
+        {
+            if (Event.current.Equals(Event.KeyboardEvent(KeyCode.D.ToString())))
+            {
+                if (_showDebugMenu)
+                {
+                    Close();
+                }
+                else
+                {
+                    Open();
+                }
+            }
+
+            if (_showDebugMenu)
+            {
+                GUI.DrawTexture(_rect, Texture2D.whiteTexture);
+                GUI.Window(0, _rect, Draw, "Debug Menu");
+            }
+            else if (_showPopup)
+            {
+                GUI.DrawTexture(_popupRect, Texture2D.whiteTexture);
+                GUI.Window(1, _popupRect, DrawPopup, "Information");
+            }
+        }
+
+        void Draw(int id)
+        {
+            if (GUILayout.Button("Open Patcher log file"))
+            {
+                OpenPatcherLogFile();
+            }
+
+            if (GUILayout.Button("Open Patcher log file location"))
+            {
+                OpenPatcherLogFileLocation();
+            }
+
+            if (GUILayout.Button("Open Launcher log file"))
+            {
+                OpenLauncherLogFile();
+            }
+
+            if (GUILayout.Button("Open Launcher log file location"))
+            {
+                OpenLauncherLogFileLocation();
+            }
+
+            if (Patcher.Instance.IsAppInstalled.Value)
+            {
+                if (GUILayout.Button("Verify all app files"))
+                {
+                    VerifyAllAppFiles();
+                }
+
+                if (GUILayout.Button("Remove all app files"))
+                {
+                    RemoveAllAppFiles();
+                }
+            }
+        }
+
+        private void VerifyAllAppFiles()
+        {
+            StartCoroutine(SetUserDecision(Patcher.UserDecision.VerifyFiles));
+        }
+
+        private void RemoveAllAppFiles()
+        {
+            StartCoroutine(SetUserDecision(Patcher.UserDecision.UninstallApp));
+        }
+
+        private IEnumerator SetUserDecision(Patcher.UserDecision userDecision)
+        {
+            while (Patcher.Instance.State.Value != PatcherState.WaitingForUserDecision)
+            {
+                yield return new WaitForSeconds(0.5f);
+            }
+
+            Patcher.Instance.SetUserDecision(userDecision);
+
+            Close();
+        }
+
+        void OpenFile(string path)
+        {
+            Close();
+            try
+            {
+                Process.Start(path);
+            }
+            catch (Exception e)
+            {
+                _graphicRaycaster.enabled = false;
```

Comment:
I don't think it is needed as `_graphicRaycaster.enabled` should be managed only by `Open` and `Close` methods.

Reply:
done